packages/product-components/src/components/SelectAssetModal/SelectAssetModal.tsx

```diff
@@ -106,33 +135,49 @@ const getNetworkCount = (options: SelectAssetOptionProps[]) => {
 
 export const SelectAssetModal = ({
     options,
-    networkCategories,
+    filterCategories,
+    searchPlaceholderText,
+    setPickedSendTokenCategory,
+    pickedSendTokenCategory = 'visibleWithBalance',
     onSelectAssetModal,
     onFavoriteClick,
+    disableSearchByNetwork = false,
     onClose,
+    selectAssetModalHeight = 'tall',
 }: SelectAssetModalProps) => {
     const intl = useIntl();
     const [search, setSearch] = useState('');
-    const [searchCategory, setSearchCategory] = useState<SelectAssetSearchCategoryType>(null); // coingeckoNativeId as fallback for ex. polygon
+    const [searchCategory, setSearchCategory] = useState<SelectAssetSearchCategory>(null); // coingeckoNativeId as fallback for ex. polygon
     const [end, setEnd] = useState(options.length);
     const data = useMemo(() => getData(options), [options]);
     const { scrollElementRef, onScroll, ShadowTop, ShadowBottom, ShadowContainer } =
         useScrollShadow();
     const networkCount = getNetworkCount(options);
 
+    const searchPlaceholder = searchPlaceholderText
+        ? searchPlaceholderText
+        : intl.formatMessage({
+              id: 'TR_SELECT_NAME_OR_ADDRESS',
+              defaultMessage: 'Search by name, symbol, network or contract address',
+          });
+
     const filteredData = data.filter(item => {
         const categoryFilter = searchCategory
             ? item.coingeckoId === searchCategory.coingeckoId ||
               item.coingeckoId === searchCategory.coingeckoNativeId
             : true;
+        const networkContractAddress = `${item.networkSymbol}${item.contractAddress ? `--${item.contractAddress}` : ''}`;
+        const contractAddress = item.contractAddress || undefined;
         const searchFor = (property: string | undefined) =>
             property?.toLocaleLowerCase().includes(search.toLocaleLowerCase());
 
         return (
             (searchFor(item.name) ||
-                searchFor(item.cryptoId) ||
                 searchFor(item.badge) ||
-                searchFor(item.symbol)) &&
+                (!disableSearchByNetwork && searchFor(item.symbol)) ||
+                (disableSearchByNetwork
+                    ? searchFor(contractAddress)
+                    : searchFor(networkContractAddress))) &&
             categoryFilter
```

Comment:
I know this is original code, but here is an opportunity to enhance:
`categoryFilter` is defined at beginning of callback, and here it is in one big `&& categoryFilter`.
:arrow_right: useless, you can early return false if `!categoryFilter` at the beginning.

Reply:
```
   const filteredData = data.filter(item => {
        if (
            searchCategory &&
            item.coingeckoId !== searchCategory.coingeckoId &&
            item.coingeckoId !== searchCategory.coingeckoNativeId
        ) {
            return false;
        }

        const contractAddress = item.contractAddress || undefined;
        const searchFor = (property: string | undefined) =>
            property?.toLocaleLowerCase().includes(search.toLocaleLowerCase());

        return (
            searchFor(item.name) ||
            searchFor(item.badge) ||
            (!disableSearchByNetwork && searchFor(item.symbol)) ||
            (disableSearchByNetwork ? searchFor(contractAddress) : searchFor(item.networkSymbol))
        );
    });
```
I believe this is it

Reply:
Yeah that's what I meant.